includes/classes/SaasOptics/SaasOptics.php

```diff
@@ -0,0 +1,139 @@
+        $user_so_contract_id = get_user_meta($user_id, 'gl_so_contract_id', true);
+        $api_key = get_option('gl_saas_optics_api_key');
+        $api_url = get_option('gl_saas_optics_api_url');
+        $date = $order->get_date_created();
+
+        if (empty($user_so_contract_id)) {
+            $billing_first_name = $order->get_billing_first_name();
+            $billing_last_name = $order->get_billing_last_name();
+            $billing_company = $order->get_billing_company();
+            $billing_address_1 = $order->get_billing_address_1();
+            $billing_address_2 = $order->get_billing_address_2();
+            $billing_city = $order->get_billing_city();
+            $billing_postcode = $order->get_billing_postcode();
+            $billing_country = $order->get_billing_country();
+            $billing_phone = $order->get_billing_phone();
+            $billing_email = $order->get_billing_email();
+            $currency = $order->get_currency();
+
+            $billing_profile = [
+                'name' => $billing_first_name . ' ' . $billing_last_name,
+                'company_name' => $billing_company,
+                "first_name" => $billing_first_name,
+                "last_name" => $billing_last_name,
+                "addr1" => $billing_address_1,
+                "addr2" => $billing_address_2,
+                "city" => $billing_city,
+                "zip_code" => $billing_postcode,
+                "country" => $billing_country,
+                "phone" => $billing_phone,
+                "email" => $billing_email,
+                "is_active" => true,
+                //"currency" => $currency,
+            ];
+
+            $customer = [
+                'name' => (empty($billing_company) ? $billing_first_name . ' ' . $billing_last_name : $billing_company),
+                'billing_profile' => $billing_profile,
+                "text_field1" => uniqid('kort_', true),
+                "is_active" => true,
+                "email" => $billing_email,
+            ];
+
+            // Create a customer object in SaasOptics
+            $client = new SaasOpticsClient($api_key, $api_url);
+            $response = $client->post('customers', ['body' => json_encode($customer)]);
+
+            $response = json_decode($response['body']);
+
+            $customer_id = $response->id;
+            update_user_meta($user_id, 'gl_so_customer_id', $customer_id);
+
+            $contract = [
+                'register' => get_option('gl_saas_optics_register', ''),
+                'billing_profile' => $billing_profile,
+                'customer' => $customer_id,
+            ];
+
+            // Create the contract object in SaasOptics related to the customer for invoicing
+            $response = $client->post('contracts', ['body' => json_encode($contract)]);
+            if ($response['response']['message'] !== 'Created') {
+                return new Exception($response['response']['message'], $response['response']['code']);
```

Comment:
Skal denne Exception sendes retur til kunden ved fejl i oprettelse? Hvis debugging (display_errors) er slået fra, vil de sandsynligvis blive præsenteret for en fejl 500